src/PropertyProcessor/Property/ConstProcessor.php

```diff
@@ -34,13 +34,17 @@ public function process(string $propertyName, JsonSchema $propertySchema): Prope
             $json['description'] ?? '',
         );
 
-        return $property
-            ->setRequired(true)
-            ->addValidator(new PropertyValidator(
-                $property,
-                '$value !== ' . var_export($json['const'], true),
-                InvalidConstException::class,
-                [$json['const']],
-            ));
+        $property->setRequired($this->propertyMetaDataCollection->isAttributeRequired($propertyName));
+
```

Comment:
Here a call to `$this->generateValidators($property, $propertySchema);` is missing. Otherwise the schema

```json
{
  "type": "object",
  "properties": {
    "property": {
      "const": "1"
    }
  },
  "required": [
    "property"
  ]
}
```

would accept an empty array as the validator related to the required flag won't be generated.

This case should also be covered by a test case where you provide no data to the `RequiredAndOptionalConstProperties.json` which should throw a `RequiredValueException`.

With that addition all cases look good!